lacework/resource_lacework_integration_ghcr.go

```diff
@@ -0,0 +1,272 @@
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"registry_notifications": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"credentials": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"password": {
+							Type:      schema.TypeString,
+							Sensitive: true,
+							Required:  true,
+						},
+						"ssl": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+					},
+				},
+			},
+
+			"limit_by_tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					StateFunc: func(val interface{}) string {
+						return strings.TrimSpace(val.(string))
+					},
+				},
+				Optional:    true,
+				Description: "A list of image tags to limit the assessment of images with matching tags",
+			},
+			"limit_by_label": {
+				Type: schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Optional:    true,
+				Description: "A list of key/value labels to limit the assessment of images",
+			},
+			"limit_by_repositories": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					StateFunc: func(val interface{}) string {
+						return strings.TrimSpace(val.(string))
+					},
+				},
+				Optional:    true,
+				Description: "A list of repositories to assess",
+			},
+			"limit_num_imgs": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  5,
+			},
+			"intg_guid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_or_updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_or_updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"org_level": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLaceworkIntegrationGhcrCreate(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	data := api.NewContainerRegistry(d.Get("name").(string),
+		api.GhcrContainerRegistry,
+		api.GhcrData{
+			LimitByTag:            castAttributeToStringSlice(d, "limit_by_tags"),
+			LimitByLabel:          castAttributeToArrayOfKeyValueMap(d, "limit_by_label"),
+			LimitByRep:            castAttributeToStringSlice(d, "limit_by_repositories"),
+			LimitNumImg:           d.Get("limit_num_imgs").(int),
+			RegistryDomain:        "ghcr",
```

Comment:
this is wrong, you should not set this field, it will be populated automatically by the SDK
```suggestion
```